src/bhattacharyya.jl

```diff
@@ -30,18 +30,36 @@ function bhattacharyya_coeff(a::AbstractVector{T}, b::AbstractVector{T}) where {
 
     sqab / sqrt(asum * bsum)
 end
+function bhattacharyya_coeff(a, b)
```

Comment:
I'm not sure, and was hoping for somebody to step in. The method above has a loop that can be `@simd`ed because it has a single simple iterator. The same optimization doesn't seem to work for loops à la `for (ai, bi) in zip(a, b)`. So it's a performance issue, which I should have commented on perhaps. For the time being, I'll rearrange that via multiple dispatch and reduce code duplication.

Reply:
Ah you're right that `@simd` can only be used for `AbstractArray`. That's a classic issue for which I hope we could find a solution.

Reply:
BTW, note that the `T <: Number` constraint could be lifted in the `AbstractArray` method.

Reply:
I refactored the `bhattacharyya` code and use multiple dispatch to distinguish the simd vs non-simd cases.